### PR TITLE
AWS Config Module

### PR DESCRIPTION
**Pull request by WOFSET**
📏 📏 📏 

This module allows you to create an AWS Config Recorder and Delivery Channel, with all the trimmings.

Please see the README.md for detailed information.

📏 📏 📏

### CODE REVIEW

File: config/config-recorder-status/config.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+resource "aws_config_configuration_recorder_status" "config-recorder" {
+  name       = "${var.name}"
+  is_enabled = "${var.is_enabled}"
```
Comment (jamesgorrie) [inline config/config-recorder-status/config.tf]: Is there a reason we're wrapping this in strings when it's a boolean type?

File: config/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Please note that this does not include creation of Config Rules, this is purely for the initial AWS Config setup.
+
+
+## Usage
+
+Here's an example of how to use the module:
+
+```hcl
+module "config-recorder" {
+  source                        = ""
```
Comment (jamesgorrie) [inline config/README.md]: Is it worth putting the `source` in [like here](https://github.com/wellcometrust/terraform-modules/blob/master/sqs/README.md)?

File: config/config-delivery-channel/config.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+resource "aws_config_delivery_channel" "config-recorder" {
+  name           = "${var.name}"
+  s3_bucket_name = "${var.s3_bucket_name}"
+  s3_key_prefix  = "${var.s3_key_prefix}"
+
```
Comment (jamesgorrie) [inline config/config-delivery-channel/config.tf]: Should we have `depends_on` [as recommended](https://www.terraform.io/docs/providers/aws/r/config_delivery_channel.html)?

File: config/config-recorder/config.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+resource "aws_config_configuration_recorder" "config-recorder" {
+  name     = "${var.name}"
+  role_arn = "${var.role_arn}"
```
Comment (jamesgorrie) [inline config/config-recorder/config.tf]: I can't see from the examples where this is coming from.
To make this all bells and whistles - should we just create the role here?

**Comment by WOFSET** [comment]
@jamesgorrie apologies for the delay getting back on this - have made changes based on the comments you previously made, have a look and let me know what you think.

I haven't added creation of the IAM role however for flexibility, eg. some accounts already have this role existing (potentially a future tidying-up exercise).

**Pull request merged**